File: linkedlist/Node.java
package linkedlist;
abstract class Node implements INode{
    protected INode child;
    private final int value;
    public Node(int value){
        this.value = value;
    }
    public Node(int value,INode child){
        this.value = value;
        this.child = child;
    }

    @Override
    public boolean contains(int val) {
        if(this.value == val) return true;
        if(this.child != null){
            return this.child.contains(val);
        }
        return false;
    }

    @Override
    public INode search(int val){
        if(this.value == val) return this;
        if(this.child != null){
            return this.child.search(val);
        }
        return null;
    }

    @Override
    public int getValue() {
        return this.value;
    }

    @Override
    public INode getChild() {
        return this.child;
    }

    @Override
    public void setChild(INode child) {
        this.child = child;
    }
}

File: src/datastructures/linkedlist/Node.php
<?php
declare(strict_types=1);
namespace ZimCodes\Self_Learning\DataStructures\LinkedList;
abstract class Node implements INode{
    public function __construct(protected mixed $value,protected ?INode $child = null){
    }
    public function getChild(): ?INode{
        return $this->child;
    }
    public function setChild(?INode $node): void{
        $this->child = $node;
    }
    public function getValue(): mixed{
        return $this->value;
    }
    public function setValue(mixed $value): void{
        $this->value = $value;
    }
    public function contains(mixed $value): bool{
        if($this->value === $value)return true;
        if($this->child !== null){
            return $this->child->contains($value);
        }
        return false;
    }
    public function search(mixed $value): ?INode{
        if($this->value === $value) return $this;
        if($this->child !== null){
            return $this->child->search($value);
        }
        return null;
    }
    abstract public function delete(mixed $value): void;
}